src/Fields/interpolate.jl

```diff
@@ -88,12 +81,32 @@ end
     return fractional_index(x, xn, Nx) 
 end
 
-@inline function fractional_x_index(x, locs, grid::LatitudeLongitudeGrid)
+@inline convert_to_0_360(x) = ((x % 360) + 360) % 360
+@inline convert_to_minus_180_180(x) = ifelse(x > 180, x - 360, x)
```

Comment:
`convert_to_0_360` works for any `x` _but_ `convert_to_minus_180_180` assumes that the input is within `[0, 360]`

```Julia
julia> convert_to_0_360(100000)
280

julia> convert_to_0_360(-100000)
80

julia> convert_to_minus_180_180(-100000)
-100000

julia> convert_to_minus_180_180(100000)
99640
```

Reply:
How about

```Julia
@inline function convert_to_minus_180_180(x)
    x = convert_to_0_360(x)
    return ifelse(x > 180, x - 360, x)
end
```

Reply:
right, I was using it after `convert_to_0_360`. But we can still embed x in `convert_to_0_360` directly inside `convert_to_minus_180_180 `